Adjust sonames on liba52 and libfaad

In debian and ubuntu, the sonames used by upstream are too tight,
because they are only provided by the development patches. Using tighter
names enables their use if only the library packages are installed.

--- a/libavcodec/liba52.c
+++ b/libavcodec/liba52.c
@@ -29,7 +29,7 @@
 
 #ifdef CONFIG_LIBA52BIN
 #include <dlfcn.h>
-static const char* liba52name = "liba52.so.0";
+static const char* liba52name = "liba52-0.7.4.so";
 #endif
 
 /**
--- a/libavcodec/libfaad.c
+++ b/libavcodec/libfaad.c
@@ -42,7 +42,7 @@
 
 #ifdef CONFIG_LIBFAADBIN
 #include <dlfcn.h>
-static const char* libfaadname = "libfaad.so";
+static const char* libfaadname = "libfaad.so.0";
 #else
 #define dlopen(a)
 #define dlclose(a)
